homeassistant/components/tplink_omada/__init__.py

```diff
@@ -0,0 +1,40 @@
+"""The TP-Link Omada integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .config_flow import OmadaHub
+from .const import DOMAIN
+
+PLATFORMS: list[Platform] = [Platform.SWITCH]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up TP-Link Omada from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    try:
+        hub = OmadaHub(hass, entry.data)
+        await hub.authenticate()
+    except Exception as ex:
```

Comment:
Don't catch a generic exception. Catch the ones you expect to come.

Reply:
Can you distinguish if it's an invalid authentication error? If so, we shouldn't raise ConfigEntryNotReady. For this PR you can `return False`. In the future you could add a `re-auth` step and raise `ConfigEntryAuthFailed` to trigger it.

Reply:
I went ahead and did this now.

Reply:
You shuold still not catch all exceptions. Instead you should catch whatever base errors Omada can raise

Reply:
OK, now catching just the Omada errors. I also changed ConnectionFailed to raise ConfigEntryNotReady, because we can assume the Omada hub will come back online later.